modules/claims_api/app/workers/claims_api/special_issue_updater.rb

```diff
@@ -0,0 +1,98 @@
+# frozen_string_literal: true
+
+require 'sidekiq'
+require 'sentry_logging'
+
+module ClaimsApi
+  class SpecialIssueUpdater
+    include Sidekiq::Worker
+    include SentryLogging
+
+    # Update special issues for a single contention/disability
+    def perform(user, contention_id, special_issues)
+      service = bgs_service(user).contention
+
+      claims = service.find_contentions_by_ptcpnt_id(user.participant_id)[:benefit_claims]
+      claim = claim_from_contention_id(claims, contention_id)
+      raise "Claim not found with contention: #{contention_id}" if claim.blank?
+
+      options = required_claim_fields(claim)
+      options = append_provided_special_issues(options, special_issues)
+
+      service.manage_contentions(options)
+    rescue BGS::ShareError, BGS::PublicError => e
+      # TODO: https://vajira.max.gov/browse/API-3277
+      log_exception_to_sentry(e)
+    end
+
+    def bgs_service(user)
+      external_key = user.common_name || user.email
+
+      BGS::Services.new(
+        external_uid: user.icn || user.uuid,
+        external_key: external_key
+      )
+    end
+
+    def append_provided_special_issues(options, special_issues)
+      options[:contentions].each do |contention|
+        next unless contention[:cntntn_id] == contention_id
```

Comment:
This is a bit confusing to me. I know there is some sort of matcher between the `disability` the api consumer sends in and the contention associated with the claim, but i'm concerned `cntntn_id` on the existing contention is incorrect.

Reply:
ok, through various discussions, this has changed to a hash of values to match on:
```
{ claim_id:, code:, name: }
```
`claim_id` is required and at least one of either `code` or `name` is required.